source/UIManager.cpp

```diff
@@ -0,0 +1,60 @@
+#include "UIManager.hpp"
+
+std::map<Ogre::String, UIOverlay *> UIManager::overlays;
+
+UIManager::UIManager(void)
+{}
+
+void UIManager::init() {
+
+	UIOverlayHUD *hud = new UIOverlayHUD();
+	Ogre::Overlay *overlayHUD = Ogre::OverlayManager::getSingleton().create("hud");
+	hud->setOverlay(overlayHUD);
+	hud->init(Ogre::OverlayManager::getSingletonPtr());
+	
+	overlays["hud"] = hud;
+
+	UIOverlayMenu *menu = new UIOverlayMenu();
+	Ogre::Overlay *overlayMenu = Ogre::OverlayManager::getSingleton().create("menu");
+	menu->setOverlay(overlayMenu);
+	menu->init(Ogre::OverlayManager::getSingletonPtr());
+	
+	overlays["menu"] = menu;
+}
+
+// PUBLIC FUNCTIONS
+
+std::vector<UIOverlay *> UIManager::getOverlays(void) {
+
+	std::vector<UIOverlay *> ov;
+	
+	std::map<Ogre::String, UIOverlay *>::iterator it;
+	for (it = overlays.begin(); it != overlays.end(); ++it) {
```

Comment:
Instead of usng a home-made for, consider using `std::vector::insert`
http://en.cppreference.com/w/cpp/container/vector/insert